### DISCUSSION TITLE
LINQ Expression for casting a primitive datatype?

### DISCUSSION THREAD

**Discussion by powellr0102** [OData/odata.net#2823] [Q&A] [answered]
The ERP software my company uses exposes all of its API endpoints as OData services. 

I'm attempting to extend the UI with a custom view that emulates how end-users are able to compose queries at runtime.

I'm trying to write a LINQ expression that will generate the following OData Filter where `ManufLeadTime` is of type `decimal` :

![image](https://github.com/OData/odata.net/assets/143731821/be3bb5b3-fc18-48ac-94e9-bb68c17448db)

The expression `x => x.ManufLeadTime.ToString().Contains("123")` throws a `NotSupportedException` and `decimal` doesn't have an explicit/implicit cast to `string`.

Is this possible or would I need to find another way to accomplish this?

Edit: These are the packages I'm using:
![image](https://github.com/OData/odata.net/assets/143731821/d5a54ee7-b37c-4dba-a286-956daae457c3)

**Comment by aboryczko** [comment, marked as answer]
I tried the following expression:
`x => ((string)((object)x.ManufLeadTime)).Contains("1");`
and it produced
`contains(cast(ManufLeadTime,Edm.String),'1')`
the client does support casting by means of `ConvertExpression`, but `ToString` is a `MethodCallExpression` which does not support `Decimal.ToString`

**Reply by powellr0102** [reply]
This worked! Thank you!
